### PR TITLE
teensy: Switch from memzip to frozen

### PR DESCRIPTION
**Pull request by dhylands**
The first commit factors out memzip into lib/memzip

The second commit switches from using memzip to using frozen, which resulted in a savings of 2472 bytes (using a set of 15 python files).

### CODE REVIEW

**Comment by pfalcon** [comment]
Looks good, thanks!

File: teensy/lexerfrozen.c [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#include <stdio.h>
+
+#include "py/lexer.h"
+
+mp_import_stat_t mp_import_stat(const char *path) {
+    return MP_IMPORT_STAT_NO_EXIST;
+}
+
+mp_lexer_t *mp_lexer_new_from_file(const char *filename)
+{
```
Comment (dpgeorge) [inline teensy/lexerfrozen.c]: Code style with brace.

File: lib/memzip/README.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
+MEMZIP - a simple readonly file system
+
+memzip takes a zip file which is comprised of uncompressed files and presents
+allows python files to be imported.
```
Comment (dpgeorge) [inline lib/memzip/README.md]: Error in wording.  Maybe "... and presents it as a filesystem, allowing Python files to be imported."

**Comment by dpgeorge** [comment]
Would it be possible to have memzip and frozen modules at the same time?  Not suggesting to do this, just wondering.

**Comment by dhylands** [comment]
Actually, I don't see any technical reason why you couldn't have them both on at the same time. It would require having an mp_import_stat and an mp_lexer_new_from_file which looked first at one and then at the other.

**Comment by dhylands** [comment]
Updated as per comments...

**Comment by dpgeorge** [comment]
Rebased and merged in 074d713bfb845a87e557e608f5215f53694e7d01 and 53ea2b5ce2f7088c871c6c264899e4eae7973859.

**Pull request closed without merging**